src/hera/workflows/script.py

```diff
@@ -566,14 +566,23 @@ def script_wrapper(
             Another callable that represents the `Script` object `__call__` method when in a Steps or DAG context,
             otherwise return the callable function unchanged.
         """
+        # instance methods are wrapped in `staticmethod`. Hera can capture that type and extract the underlying
```

Comment:
Something to keep in mind/we should investigate a bit - how are wrapped functions treated in general?